File: net/split/Split_PDF.cs
using Aspose.Pdf;
using Aspose.Pdf.Facades;

namespace aspose_snippets.net
{
    public static partial class Split
    {
        public static void PDF()
        {
            var pathSource = @"..\..\TestData\test.pdf";
            var pdfEditor = new PdfFileEditor();
            int beg = 1, end = 1;

            using (var fs = new FileStream(pathSource, FileMode.Open, FileAccess.Read))
            {
                
                using (var doc = new Document(fs))
                {
                    end = doc.Pages.Count;
                }
            }

            if(end > 1)
            {
                    end /= 2;
            }

            pdfEditor.Extract(pathSource, beg, end, "./half.pdf");
        }
    }
}

File: java/split/src/aspose_snippets/java/Split_PDF.java
package aspose_snippets.java;

public class Split_PDF {
    public static void Execute() {
        String pathSource = "../../TestData/test.pdf";
        com.aspose.pdf.facades.PdfFileEditor pdfEditor = new com.aspose.pdf.facades.PdfFileEditor();
        int beg = 1, end = 1;
    
        com.aspose.pdf.Document doc = new com.aspose.pdf.Document(pathSource);
        end = doc.getPages().size();
    
        if(end > 1)
        {
            end /= 2;
        }
    
        pdfEditor.extract(pathSource, beg, end, "pdf_half.pdf");
    }
}
